samples/excel/worksheet/add-delete-rename-move-worksheet.yaml

```diff
@@ -0,0 +1,218 @@
+id: ''
+gist: ''
+source: EXCEL
+author: Microsoft
+name: 'Add, delete, rename and move worksheet'
+description: 'Explore the APIs to add, delete, rename and change the position of a worksheet'
+script:
+  content: |-
+    $("#add-worksheet").click(addWorksheet);
+    $("#delete-worksheet").click(deleteWorksheet);
+    $("#rename-worksheet").click(renameWorksheet);
+    $("#move-worksheet").click(moveWorksheet);
+
+    async function addWorksheet() {
+        try {
+            await Excel.run(async (context) => {
+                let sheets = context.workbook.worksheets;
+
+                let sheet = sheets.add();
+                sheet.load("name, position");
+                
+                await context.sync();
+
+                console.log(`Added worksheet named "${sheet.name}" in position ${sheet.position}`)
+            });
+
+            console.log("Done!");
+        }
+        catch (error) {
+            OfficeHelpers.Utilities.log(error);
+        }
+    }
+
+    async function deleteWorksheet() {
+        try {
+            await Excel.run(async (context) => {
+                let sheets = context.workbook.worksheets;
+                sheets.load("items/name");
+
+                await context.sync();
+
+                if (sheets.items.length > 1) {
+                    let lastSheet = sheets.items[sheets.items.length - 1];
+
+                    console.log(`Deleting worksheet named "${lastSheet.name}"...`); 
+                    lastSheet.delete();
+
+                    await context.sync();
+                    
+                } else {
+                    console.log("Unable to delete the last worksheet in the workbook");
+                }    
+            });
+
+            console.log("Done!");
+        }
+        catch (error) {
+            OfficeHelpers.Utilities.log(error);
+        }
+    }
+
+    async function renameWorksheet() {
+        try {
+            await Excel.run(async (context) => {
+                let currentSheet = context.workbook.worksheets.getActiveWorksheet();
+
+                currentSheet.name = await uniqueWorksheetName(context);            
```

Comment:
This seems exceedingly difficult.  Why not either add a text box for the name, or just hardcode one and let it fail if the name already exists.  That seems much more realistic.